core/domain/stats_services.py

```diff
@@ -264,21 +269,31 @@ def advance_version_of_exp_stats(
 
         return exp_stats
 
+    state_name_stats_mapping = {}
+    old_state_names = utils.compute_list_difference(
+        exp_stats.state_stats_mapping,
+        exp_versions_diff.deleted_state_names)
+
     # Handling state deletions.
-    for state_name in exp_versions_diff.deleted_state_names:
-        exp_stats.state_stats_mapping.pop(state_name)
+    for state_name in old_state_names:
```

Comment:
In what way does this handle state deletions?

Reply:
Ah sorry, it just copies over state names from the prev version that weren't modified. Will clarify in the comment.

Reply:
Done.